lib/ansible/plugins/lookup/first_found.py

```diff
@@ -106,71 +110,100 @@
 
 from jinja2.exceptions import UndefinedError
 
-from ansible.errors import AnsibleFileNotFound, AnsibleLookupError, AnsibleUndefinedVariable
+from ansible.errors import AnsibleLookupError, AnsibleUndefinedVariable
+from ansible.module_utils.common._collections_compat import Mapping, Sequence
 from ansible.module_utils.six import string_types
-from ansible.module_utils.parsing.convert_bool import boolean
 from ansible.plugins.lookup import LookupBase
 
 
+def _split_on(terms, spliters=','):
+
+    # TODO: fix as it does not allow spaces in names
+    termlist = []
+    if isinstance(terms, string_types):
+        for spliter in spliters:
+            terms = terms.replace(spliter, ' ')
+        termlist = terms.split(' ')
+    else:
+        # added since options will already listify
+        for t in terms:
+            termlist.extend(_split_on(t, spliters))
+
+    return termlist
+
+
 class LookupModule(LookupBase):
 
-    def run(self, terms, variables, **kwargs):
+    def _process_terms(self, terms, variables, kwargs):
 
-        anydict = False
+        total_search = []
         skip = False
 
+        # can use a dict instead of list item to pass inline config
         for term in terms:
-            if isinstance(term, dict):
-                anydict = True
+            if isinstance(term, Mapping):
+                self.set_options(var_options=variables, direct=term)
+            elif isinstance(term, string_types):
+                self.set_options(var_options=variables, direct=kwargs)
```

Comment:
Shouldn't this only be called if the set_options in line 145 was never called? Otherwise strings behind a dictionary entry would overwrite the options provided in the dictionary.

Reply:
yes ... i made comments to that effect, but this is only way i saw of keeping backwards compat